torch/_inductor/fx_passes/post_grad.py

```diff
@@ -1070,19 +1079,29 @@ def move_constructors_to_cuda(graph):
 
         constructors.append(node)
 
-    # not handling multiple cuda devices initially
-    if not constructors or len(cuda_devices) != 1:
-        return
+    return constructors, cuda_devices
 
+
+def move_constructors_to_cuda_impl(graph_module):
+    graph = graph_module.graph
+    constructors, cuda_devices = gather_constructors_and_cuda_devices(graph)
     movable_constructors = find_movable_constructors(graph, constructors)
 
+    if len(movable_constructors) == 0:
+        return
+
+    if len(cuda_devices) != 1:
+        raise ZeroOrMultipleDevicesError("cuda", cuda_devices)
+
+    cuda_device = next(iter(cuda_devices))
+
     for node in movable_constructors:
         kwargs = node.kwargs.copy()
-        kwargs["device"] = next(iter(cuda_devices))
+        kwargs["device"] = cuda_device
         node.kwargs = kwargs
 
 
-def find_movable_constructors(graph, constructors: List[fx.Node]) -> Set[fx.Node]:
+def find_movable_constructors(graph, constructors: List[fx.Node]) -> List[fx.Node]:
```

Comment:
Can we keep this as a set? it makes more clear that the output entries will be unique